### ISSUE TITLE
[BUG] [JSL] "core_mask is not in host managed cores" during firmware booting

### ISSUE THREAD

**Issue by ghost** [thesofproject/linux#3274]
**Describe the bug**
In the Jaspherlake machine, the driver will print kernel log like `error: core_mask is not in host managed cores` after `FW_READY` IPC is done. I think this is introduced by <https://github.com/thesofproject/linux/pull/3272>.

**To Reproduce**
Check dmesg after kernal boot.

**Reproduction Rate**
100%

**Expected behavior**
No error in the kernel messages.

**Impact**
It seems this error didn't block normal audio functions, but the *error level* logs will break our test cases.

**Environment**
Kernel Branch: topic/sof-dev
Kernel Commit: 66bd4e2a
SOF Branch: main
SOF Commit: https://github.com/thesofproject/sof/commit/94517163580d
Topology: sof-jsl-nocodec.tplg
Platform: JSL_RVP_NOCODEC

**console output**
*dmesg*:

```
[    4.130743] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Firmware download successful, booting...
[    4.138968] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: ipc rx: 0x70000000: FW_READY
[    4.138975] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: ipc: DSP is ready 0x70000000 offset 0x81000
[    4.139057] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Firmware info: version 1:9:0-94517
[    4.139061] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Firmware: ABI 3:20:0 Kernel ABI 3:18:0
[    4.139065] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: warn: FW ABI is more recent than kernel
[    4.139067] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Firmware debug build 65535 on dtermin.-fwready. - options:
                        GDB: disabled
                        lock debug: disabled
                        lock vdebug: disabled
[    4.139164] kernel: sof-audio-pci-intel-icl 0000:00:1f.3:  mailbox upstream 0x81000 - size 0x1000
[    4.139167] kernel: sof-audio-pci-intel-icl 0000:00:1f.3:  mailbox downstream 0xa0000 - size 0x2000
[    4.139170] kernel: sof-audio-pci-intel-icl 0000:00:1f.3:  stream region 0xc1000 - size 0x1000
[    4.139173] kernel: sof-audio-pci-intel-icl 0000:00:1f.3:  debug region 0xc0000 - size 0x800
[    4.139177] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: fw_state change: 2 -> 5
[    4.139185] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: ipc rx done: 0x70000000: FW_READY
[    4.139192] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: error: core_mask is not in host managed cores
[    4.139235] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Debug PCIR: 00000000 at  00000048
[    4.139242] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Debug PCIW: 00000002 at  00000048
[    4.139253] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Debug PCIR: 00000004 at  00000044
[    4.139258] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: Debug PCIW: 00000000 at  00000044
[    4.139260] kernel: sof-audio-pci-intel-icl 0000:00:1f.3: firmware boot complete
```

**Comment by plbossart** [comment]
@ranj063 I have a doubt on this one. 

Can JasperLake really use the ICL ops if it has only 2 cores? IIRC we used core3 for the power-up sequences on ICL. So maybe using CNL ops is a feature, not a bug?

**Comment by ranj063** [comment]
@plbossart I think the problem is that ICL uses ICCMAX workaround but JSL doesnt need it. So maybe using the CNL ops might have been a feature

**Comment by miRoox** [comment]
closed by #3276
